agent-control/src/agent_type/runtime_config/on_host/package.rs

```diff
@@ -0,0 +1,140 @@
+use crate::agent_type::definition::Variables;
+use crate::agent_type::error::AgentTypeError;
+use crate::agent_type::runtime_config::templateable_value::TemplateableValue;
+use crate::agent_type::templates::Templateable;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, Default, Clone, PartialEq)]
+pub(super) struct Package {
+    /// Package type (tar.gz, zip, etc).
+    pub(super) r#type: String, // Using `r#type` to avoid keyword conflict
+
+    /// Download defines the supported repository sources for the packages.
+    pub(super) download: Download,
+    //TODO: implement signature, install and uninstall fields when defined.
+}
+
+#[derive(Debug, Deserialize, Default, Clone, PartialEq)]
+pub struct Download {
+    /// OCI repository definition
+    pub oci: Oci,
+}
+
+/// See [`AgentDirectoryEntry`] and [`crate::agent_type::runtime_config::on_host::filesystem::DirEntriesType`] for more details.
+#[derive(Debug, Default, Deserialize, Clone, PartialEq)]
+pub struct OCIVersion(String);
+
+#[derive(Debug, Deserialize, Default, Clone, PartialEq)]
+pub struct Oci {
+    /// OCI registry url.
+    pub registry: TemplateableValue<String>,
+    /// Repository name.
+    pub repository: TemplateableValue<String>,
+    /// Package version including tag, digest or tag + digest.
+    /// It prepends ":" if it's a tag or tag + digest
+    #[serde(default)]
+    pub version: TemplateableValue<OCIVersion>,
+}
+
+impl Templateable for OCIVersion {
+    type Output = Self;
+    fn template_with(self, variables: &Variables) -> Result<Self::Output, AgentTypeError> {
+        let version = self.0.template_with(variables)?;
+        if version.starts_with('@') || version.is_empty() {
```

Comment:
not sure but i think that the digest format is something like <algo>-<digest>

Reply:
I've modified this in last commit, anyway, the digest can be whatever string as long as it starts by @

Reply:
you're right  i guess that there would be no great improvement , just some early error , consider it a nit!